src/services/stepsService.ts

```diff
@@ -335,30 +335,32 @@ export class StepsService {
    * @param targetNode
    * @param newStep
    */
-  handleInsertStep(targetNode: IVizStepNode | undefined, newStep: IStepProps) {
-    if (targetNode?.data.branchInfo) {
-      const currentStepNested = this.getStepNested(targetNode.data.step);
+  handleInsertStep(targetNode: IVizStepNode | undefined, step: IStepProps) {
+    return fetchStepDetails(step.id).then((newStep) => {
+      if (targetNode?.data.branchInfo) {
+        const currentStepNested = this.getStepNested(targetNode.data.step);
 
-      if (currentStepNested) {
-        const stepsCopy = this.integrationJsonStore.integrationJson.steps.slice();
-        let newParentStep = getDeepValue(stepsCopy, currentStepNested.pathToParentStep);
-        const newBranch = newParentStep.branches[currentStepNested.branchIndex];
-        const targetIdx = this.findStepIdxWithUUID(targetNode.data.step.UUID, newBranch.steps);
-        newParentStep.branches[currentStepNested.branchIndex].steps = StepsService.insertStep(
-          newBranch.steps,
-          targetIdx,
-          newStep
-        );
-
-        this.integrationJsonStore.replaceBranchParentStep(
-          newParentStep,
-          currentStepNested.pathToParentStep
-        );
+        if (currentStepNested) {
+          const stepsCopy = this.integrationJsonStore.integrationJson.steps.slice();
+          let newParentStep = getDeepValue(stepsCopy, currentStepNested.pathToParentStep);
+          const newBranch = newParentStep.branches[currentStepNested.branchIndex];
+          const targetIdx = this.findStepIdxWithUUID(targetNode.data.step.UUID, newBranch.steps);
```

Comment:
@mmelko it's difficult to see what has been changed in this section, could it be something with the code formatting?

Reply:
Nevermind I see it was wrapped with an `if/else`